### PR TITLE
To merge two Keywords(Source Keyword and Target Keyword)

### PR DESCRIPTION
**Pull request by Veda-Gogineni**
addresses #885 
* `keyword.merge(other)` will update references (Assets, Annotations) from _other_ to _self_, then delete _other_
* api endpoint `/api/v1/keywords/OTHER_GUID/GUID/merge` will do this merge

### CODE REVIEW

File: app/modules/keywords/models.py [hunk 1]
```diff
@@ -117,3 +117,38 @@ def number_referenced_dependencies(self):
...
+        for ref_ann in ref_annos:
+            ref_ann.keyword_guid = self.guid
+
+        # Get and update all the assets that reference the other keyword
+        ref_assets = AssetTags.query.filter_by(tag_guid=other.guid).all()
+        for ref_asset in ref_assets:
+            ref_asset.tag_guid = self.guid
+
+        # Delete the other keyword
+        other.delete()
```
Comment (naknomum) [inline app/modules/keywords/models.py:145]: i _think_ we would want to move this until **after** the annotation updates below? i am pretty sure the `.delete()` will remove the (source) keywords off of Annotations below.

File: app/modules/keywords/models.py [hunk 2]
```diff
@@ -117,3 +117,38 @@ def number_referenced_dependencies(self):
...
+
+        try:
+            updated_ref_ann = AnnotationKeywords.query.filter_by(keyword_guid=self.guid).first()
```
Comment (naknomum) [inline app/modules/keywords/models.py:148]: there can be **multiple** Annotations which have this keyword needing to be re-assigned. so it should follow the pattern above like on Assets: do a `for ... in` iteration over `.all()` Annotations.

File: app/modules/keywords/models.py (hunk 1 above)
Comment (Veda-Gogineni) [reply app/modules/keywords/models.py:145]: While testing when I refer annotation to keyword.  The changes to the annotations and assets are being tracked and hence are committed when other.delete() is called, as it internally calls db.session.commit()

**Review by naknomum — APPROVED** [review]
looks good

**Pull request merged**